src/linux.rs

```diff
@@ -158,6 +158,16 @@ fn local_ip_impl(family: RtAddrFamily) -> Result<IpAddr, Error> {
     let mut netlink_socket = NlSocketHandle::connect(NlFamily::Route, None, &[])
         .map_err(|err| Error::StrategyError(err.to_string()))?;
 
+    match local_ip_impl_route(family, &mut netlink_socket) {
+        Ok(ip_addr) => Ok(ip_addr),
+        Err(_error) => local_ip_impl_addr(family, &mut netlink_socket),
```

Comment:
Can we check here for the error to be "LocalIpAddressNotFound"? That way we have a controlled flow for the fallback, allowing us to reduce ambiguity. 

```suggestion
        Err(error) if error == Error::LocalIpAddressNotFound => local_ip_impl_addr(family, &mut netlink_socket),
        Err(e) => Err(e)
```

Reply:
Yes, that's better!